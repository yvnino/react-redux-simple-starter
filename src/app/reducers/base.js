import { set } from 'lodash/fp';
import AT from 'constants/actionTypes';

export const initialState = {
  text: null
};

const baseReducer = (state = initialState, action) => {
  switch (action.type) {
    case AT.SET_TEXT_ASYNC_SUCCESS:
    case AT.SET_TEXT_SYNC: {
      return set('text', action.payload, state);
    }
    case AT.SET_TEXT_ASYNC_FAIL: {
      return set('text', action.payload, state);
    }
    default:
      return state;
  }
};

export default baseReducer;
