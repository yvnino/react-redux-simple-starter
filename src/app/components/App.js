import React from 'react';
import PropTypes from 'prop-types';
import Button from '@material-ui/core/Button';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import * as BaseActions from '../actions/BaseActions';

export class App extends React.Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  handleSyncClick = () => {
    this.props.actions.setTextSync('text');
  }

  handleAsyncClick = () => {
    this.props.actions.setTextAsync();
  }

  render() {
    return (
      <div>
        <div style={ { color: '#fff' } } >Loaded!</div>
        <div>
          <Button variant="contained" color="secondary" id="SetTextSyncButton" onClick={ this.handleSyncClick }>
            SetTextSync
          </Button>
        </div>
        <div>
          <Button variant="contained" color="primary" id="SetTextAsyncButton" onClick={ this.handleAsyncClick }>
            SetTextAsync
          </Button>
        </div>
        <div style={ { color: '#fff' } }>
          {this.props.text}
        </div>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    text: state.baseReducer.text
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    actions: bindActionCreators({ ...BaseActions }, dispatch)
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(App);

App.defaultProps = {
  text: null
};

App.propTypes = {
  actions: PropTypes.object.isRequired,
  text: PropTypes.string
};

