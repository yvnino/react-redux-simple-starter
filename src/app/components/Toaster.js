
import React from 'react';
import PropTypes from 'prop-types';
import AlertContainer from 'react-alert';
import { connect } from 'react-redux';
import Ionicon from 'react-ionicons';

class Toaster extends React.Component {
  constructor(props) {
    super(props);
    this.showAlertSuccess = this.showAlertSuccess.bind(this);
    this.showAlertFail = this.showAlertFail.bind(this);
    this.msg = null;
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.toasterState.showSuccessMsg) {
      this.showAlertSuccess(nextProps.toasterState.msg);
    } else if (nextProps.toasterState.showFailMsg) {
      this.showAlertFail(nextProps.toasterState.msg);
    }
  }

  showAlertSuccess(msg) {
    const toBeat = true;

    this.msg.show(msg, {
      time: 4000,
      type: 'success',
      icon: <Ionicon icon="md-checkmark" color="#5cb85c" fontSize="35px" beat={ toBeat } />
    });
  }
  showAlertFail(msg) {
    const toBeat = true;
    
    this.msg.show(msg, {
      time: 20000,
      type: 'error',
      icon: <Ionicon icon="ios-close-circle-outline" color="#d9534f" fontSize="35px" beat={ toBeat } />
    });
  }

  render() {
    const alertOptions = {
      offset: 14,
      position: 'top right',
      theme: 'light',
      time: 5000,
      transition: 'scale'
    };

    return (
      <div className="Toaster">
        <AlertContainer ref={ (a) => { this.msg = a; } } { ...alertOptions } />
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    toasterState: state.toasterReducer
  };
};

Toaster.defaultProps = {
  toasterState: {}
};

Toaster.propTypes = {
  toasterState: PropTypes.object
};


const toaster = connect(mapStateToProps)(Toaster);

export default toaster;
