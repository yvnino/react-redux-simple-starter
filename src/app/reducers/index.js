import { combineReducers } from 'redux';
import baseReducer from './base';

const rootReducer = combineReducers({
  baseReducer
});

export default rootReducer;
