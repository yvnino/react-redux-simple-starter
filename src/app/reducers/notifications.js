const toasterReducer = (state = { showSuccessMsg: false, msg: '' }, action) => {
  switch (action.type) {
    case 'SUCCESS_MESSAGE': {
      return Object.assign({}, {
        showSuccessMsg: true,
        msg: action.data.msg
      });
    }
    case 'FAIL_MESSAGE': {
      return Object.assign({}, {
        showFailMsg: true,
        msg: action.e.msg
      });
    }
    default:
      return state;
  }
};

export default toasterReducer;
