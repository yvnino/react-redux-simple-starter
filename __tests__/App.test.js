import React from 'react';
import jest from 'jest-mock';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, { shallow } from 'enzyme';
import  {App}  from '../src/app/components/App';
import {initialState as baseReducerInitialState} from '../src/app/reducers/base';
// React 16 Enzyme adapter
Enzyme.configure({ adapter: new Adapter() });

const setTextAsync = jest.fn();
const setTextSync = jest.fn();

const actionsMock = {
  setTextSync,
  setTextAsync
};
const setup = (props = {}) => ({
  ...props,
  actions: actionsMock,
  ...baseReducerInitialState
});

describe('<App />', () => {
  const props = setup();
  let app = shallow(<App {...props}  />);
  
  
  describe('basic', () => {
    it('render', () => {
      expect(App).toMatchSnapshot();
    });

    it('initialized `state`', () => {
      expect(app.state()).toEqual({});
    });

    it('initialized props', () => {
      const { actions, text } = app.instance().props;
      expect(text).toEqual(null);
      expect(actions).toEqual(actionsMock);
    });
  });

  describe('buttons', () => {
    beforeEach(() => {
      app.setProps({
        text: null
      })
    });

    it('test setTextSync', () => {
      app.find('#SetTextSyncButton').simulate('click');
      expect(actionsMock.setTextSync).toBeCalled();
    });

    it('test setTextAsync', () => {
      app.find('#SetTextAsyncButton').simulate('click');
      expect(actionsMock.setTextAsync).toBeCalled();
    });
  });
});
