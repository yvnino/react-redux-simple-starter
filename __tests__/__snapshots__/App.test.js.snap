// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> basic render 1`] = `[Function]`;
